src/coreclr/tools/aot/ILCompiler.ReadyToRun/IL/ReadyToRunILProvider.cs

```diff
@@ -130,5 +185,93 @@ public override MethodIL GetMethodIL(MethodDesc method)
                 return null;
             }
         }
+
+        /// <summary>
+        /// A MethodIL Provider which provides tokens relative to a MutableModule. Used to implement cross
+        /// module inlining of code in ReadyToRun files.
+        /// </summary>
+        class ManifestModuleWrappedMethodIL : MethodIL, IEcmaMethodIL, IMethodTokensAreUseableInCompilation
+        {
+            int _maxStack;
+            bool _isInitLocals;
+            MethodDesc _owningMethod;
+            ILExceptionRegion[] _exceptionRegions;
+            byte[] _ilBytes;
+            LocalVariableDefinition[] _locals;
+
+            MutableModule _mutableModule;
+
+            public ManifestModuleWrappedMethodIL(MutableModule mutableModule, EcmaMethodIL wrappedMethod)
+            {
+                mutableModule.TryGetEntityHandle(wrappedMethod.OwningMethod);
+                _mutableModule = mutableModule;
+                _maxStack = wrappedMethod.MaxStack;
+                _isInitLocals = wrappedMethod.IsInitLocals;
+                _owningMethod = wrappedMethod.OwningMethod;
+                _exceptionRegions = (ILExceptionRegion[])wrappedMethod.GetExceptionRegions().Clone();
+                _ilBytes = (byte[])wrappedMethod.GetILBytes().Clone();
+                _locals = (LocalVariableDefinition[])wrappedMethod.GetLocals();
+
+                for (int i = 0; i < _exceptionRegions.Length; i++)
+                {
+                    var region = _exceptionRegions[i];
+                    if (region.Kind == ILExceptionRegionKind.Catch)
+                    {
+                        var newHandle = _mutableModule.TryGetHandle((TypeSystemEntity)wrappedMethod.GetObject(region.ClassToken));
+                        if (!newHandle.HasValue)
+                        {
+                            throw new Exception("Nope");
+                        }
+                        _exceptionRegions[i] = new ILExceptionRegion(region.Kind, region.TryOffset, region.TryLength, region.HandlerOffset, region.HandlerLength, newHandle.Value, newHandle.Value);
+                    }
+                }
+
+                ILTokenReplacer.Replace(_ilBytes, GetMutableModuleToken);
+#if DEBUG
+                Debug.Assert(ReadyToRunStandaloneMethodMetadata.Compute((EcmaMethod)_owningMethod) != null);
+#endif // DEBUG
+
+                return;
+
+                int GetMutableModuleToken(int token)
+                {
+                    object result = wrappedMethod.GetObject(token);
+                    int? newToken;
+                    if (result is string str)
+                    {
+                        newToken = mutableModule.TryGetStringHandle(str);
+                    }
+                    else
+                    {
+                        newToken = mutableModule.TryGetHandle((TypeSystemEntity)result);
+                    }
+                    if (!newToken.HasValue)
+                    {
+                        throw new Exception("Nope2");
```

Comment:
I've replaced it with a pattern where if this failure occurs, then the method is silently dropped. This will be needed for future work where we expand the set of cross module compilable methods beyond just that of System.Private.CoreLib